### PR TITLE
Add extra checks in DMA verifiers

### PR DESCRIPTION
**Pull request by AndraBisca**

### CODE REVIEW

File: lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp [hunk 1]
```diff
@@ -92,40 +92,20 @@ class DMAChannelAnalysis {
...
       masterChannelsPerTile[tile]++;
-    }
     DMAChannel dmaChan = {DMAChannelDir::MM2S, masterChannelsPerTile[tile]};
     return dmaChan;
   }
 
   /// Given an AIE tile, returns its next usable slave channel.
   DMAChannel getSlaveDMAChannel(Value tile) {
-    if (slaveChannelsPerTile.find(tile) == slaveChannelsPerTile.end()) {
+    if (slaveChannelsPerTile.find(tile) == slaveChannelsPerTile.end())
```
Comment (makslevental) [inline lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp:105]: yay :)

File: lib/Dialect/AIE/IR/AIEDialect.cpp [hunk 2]
```diff
@@ -1436,6 +1497,14 @@ LogicalResult MemTileDMAOp::verify() {
...
   }
 
+  if ((int)inputChannels.size() >
+      getTileOp().getNumSourceConnections(WireBundle::DMA))
+    return emitOpError("uses more input channels than available on this tile");
+
+  if ((int)outputChannels.size() >
+      getTileOp().getNumDestConnections(WireBundle::DMA))
+    return emitOpError("uses more output channels than available on this tile");
+
```
[the remark this answers is not in the document]
Comment (AndraBisca) [reply lib/Dialect/AIE/IR/AIEDialect.cpp:1507]: I will take a look to see how much would have to change if I changed the return type to `size_t`!

File: lib/Dialect/AIE/Transforms/AIEFindFlows.cpp [hunk 3]
```diff
@@ -233,7 +233,7 @@ static void findFlowsFrom(TileOp op, ConnectivityAnalysis &analysis,
     LLVM_DEBUG(llvm::dbgs()
                << op << stringifyWireBundle(bundle) << " has "
                << op.getNumSourceConnections(bundle) << " Connections\n");
-    for (int i = 0; i < op.getNumSourceConnections(bundle); i++) {
+    for (int i = 0; i < (int)op.getNumSourceConnections(bundle); i++) {
```
Comment (makslevental) [inline lib/Dialect/AIE/Transforms/AIEFindFlows.cpp:236]: Missed one :)

**Review by makslevental — APPROVED** [review]
I'm glad the `size_t` change wasn't such a big deal.

**Pull request merged**